Resources/frontend/js/jquery.adyen-payment-selection.js

```diff
@@ -358,7 +350,7 @@
          */
         __isGiftCardType: function (paymentType) {
             var me = this;
-            var paymentGroups = me.opts.adyenPaymentMethodsResponse['groups'] || [];
+            var paymentGroups = me.adyenConfiguration.paymentMethodsResponse['groups'] || [];
```

Comment:
Where is the `groups` index? Is this the /paymentMethods response? It seems like there's no `groups` index there for v67 https://docs.adyen.com/api-explorer/#/CheckoutService/v67/post/paymentMethods.

v52 does have it https://docs.adyen.com/api-explorer/#/CheckoutService/v52/post/paymentMethods, but the plugin is already using v67.

Reply:
Good catch, this needs a closes look and a refactor